src/main/scala/org/broadinstitute/dsde/workbench/avram/util/SlickDatabaseFactory.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import com.zaxxer.hikari.HikariDataSource
+import org.apache.commons.dbcp2.BasicDataSource
+import org.broadinstitute.dsde.workbench.avram.config.DbcpDataSourceConfig
+import slick.jdbc.PostgresProfile.api._
+
+/**
+  * Provides access to a slick Database object using the provided configuration.
+  *
+  * @param dataSourceConfig database and connection pool configuration
+  */
+class SlickDatabaseFactory(dataSourceConfig: DbcpDataSourceConfig) {
+
+  // See https://commons.apache.org/proper/commons-dbcp/configuration.html for configuration options and defaults
```

Comment:
Maybe lift this into a `makeDbcpDataSource` method? (similar to `makeHikariDataSource`)

Reply:
Good call. `makeHikariDataSource` was mostly a cheat to keep unused code in the commit while removing conceptual complexity for the reviewer.